From 311f25363e35f5058fc8a8f1ba9c0835f169fd96 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 3 Apr 2014 14:41:42 +0200
Subject: [PATCH 4/4] add workaround for problem in gdb

---
 complete.c | 2 +-
 readline.h | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/complete.c b/complete.c
index cd9aebe..35d86e9 100644
--- a/complete.c
+++ b/complete.c
@@ -518,7 +518,7 @@ get_y_or_n (for_pager)
      driven functions.  Have to wait until next major version to add new
      state definition, since it will change value of RL_STATE_DONE. */
 #if defined (READLINE_CALLBACKS)
-  if (RL_ISSTATE (RL_STATE_CALLBACK))
+  if (RL_ISSTATE (RL_STATE_CALLBACK) && (! RL_ISSTATE (RL_STATE_FEDORA_GDB)))
     return 1;
 #endif
 
diff --git a/readline.h b/readline.h
index 08dcd2b..1df0639 100644
--- a/readline.h
+++ b/readline.h
@@ -870,6 +870,8 @@ extern int rl_inhibit_completion;
 
 #define RL_STATE_DONE		0x1000000	/* done; accepted line */
 
+#define RL_STATE_FEDORA_GDB		0x2000000	/* exception for fedora gdb */
+
 #define RL_SETSTATE(x)		(rl_readline_state |= (x))
 #define RL_UNSETSTATE(x)	(rl_readline_state &= ~(x))
 #define RL_ISSTATE(x)		(rl_readline_state & (x))
-- 
1.8.5.3

