From 0415e19a3f41596800e6144a5c5acca05929a90a Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 17 Feb 2010 21:15:01 +0100
Subject: [PATCH] Fix the version number

---
 readline.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/readline.h b/readline.h
index 7a4ffaa..9a9e18a 100644
--- a/readline.h
+++ b/readline.h
@@ -39,9 +39,9 @@ extern "C" {
 #endif
 
 /* Hex-encoded Readline version number. */
-#define RL_READLINE_VERSION	0x0600		/* Readline 6.0 */
+#define RL_READLINE_VERSION	0x0601		/* Readline 6.0 */
 #define RL_VERSION_MAJOR	6
-#define RL_VERSION_MINOR	0
+#define RL_VERSION_MINOR	1
 
 /* Readline data structures. */
 
-- 
1.7.0

